source/loader/layers/sanitizer/msan/msan_report.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+ * See LICENSE.TXT
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * @file msan_report.cpp
+ *
+ */
+
+#include "msan_report.hpp"
+#include "msan_libdevice.hpp"
+
+#include "sanitizer_common/sanitizer_common.hpp"
+#include "sanitizer_common/sanitizer_utils.hpp"
+#include "ur_sanitizer_layer.hpp"
+
+namespace ur_sanitizer_layer {
+namespace msan {
+
+void ReportUsesUninitializedValue(const MsanErrorReport &Report,
+                                  ur_kernel_handle_t Kernel) {
+    const char *File = Report.File[0] ? Report.File : "<unknown file>";
+    const char *Func = Report.Func[0] ? Report.Func : "<unknown func>";
+    auto KernelName = GetKernelName(Kernel);
+
+    // Try to demangle the kernel name
+    KernelName = DemangleName(KernelName);
+
+    getContext()->logger.always(
+        "====WARNING: MemorySanitizer: use-of-uninitialized-value");
```

Comment:
If we use Memory sanitizer here, should we update the DevASAN? To speak for myself, I think we can use "DeviceSanitizer" for all sanitizers.